diff -urp qstat-2.11-orig/Makefile.noauto qstat-2.11/Makefile.noauto
--- qstat-2.11-orig/Makefile.noauto	2006-10-28 14:36:58.000000000 +0200
+++ qstat-2.11/Makefile.noauto	2009-05-21 20:57:19.235051200 +0200
@@ -5,7 +5,8 @@
 #LDLIBS  =
 
 #CFLAGS += -Dsysconfdir=\"/etc\"
-CFLAGS = -DDEBUG=1
+#CFLAGS = -DDEBUG=1
+CFLAGS = -DNDEBUG /MD /O2 /nologo
 
 ## NOTE: if you get errors when linking qstat (missing symbols or
 ## libraries), then modify LDFLAGS or LDLIBS
@@ -15,7 +16,7 @@ OBJ = $(SRC:.c=.obj)
 O   = $(SRC:.c=.o)
 
 SOLARIS_LIBS = -lsocket -lnsl
-WINDOWS_LIBS = /ML wsock32.lib
+WINDOWS_LIBS = wsock32.lib
 OS2_LIBS     = so32dll.lib tcp32dll.lib
 EMX_LIBS     = -lsocket
 
diff -urp qstat-2.11-orig/qstat.c qstat-2.11/qstat.c
--- qstat-2.11-orig/qstat.c	2006-10-28 14:37:18.000000000 +0200
+++ qstat-2.11/qstat.c	2009-05-21 20:41:54.585470400 +0200
@@ -586,8 +586,11 @@ display_server( struct qserver *server)
     if ( player_sort)
 	sort_players( server);
 
-    if ( raw_display)
+    if ( raw_display)  {
 	raw_display_server( server);
+	if ( OF == stdout)
+	    fflush( stdout);
+    }
     else if ( xml_display)
 	xml_display_server( server);
     else if ( have_server_template())
